src/pipeline_v3/utils/config.py

```
@@ -114,6 +114,37 @@ class SearchSettings:
+@dataclass
+class ProcessingProfile:
+    """Defines a reusable processing configuration profile."""
+    document_type: str = "auto"
+    processing_options: List[str] = field(default_factory=list)
+    timeout_multiplier: float = 1.0  # Multiply default timeout
+    description: str = ""
+
+@dataclass
+class ProcessingProfileSettings:
+    """Collection of named processing profiles."""
+    profiles: Dict[str, ProcessingProfile] = field(default_factory=lambda: {
```

Ruff (0.11.9)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)